File: Project Playground source code/Assets/Scripts/PauseMenu.cs
using UnityEngine;
using System.Collections;

public class PauseMenu : MonoBehaviour 
	{
		public GUISkin skin;
		public bool paused = false;
	
	void Update(){
		if(paused){
		Time.timeScale = 0;	
		}
		else{
		Time.timeScale = 1;
		}
		
		if(Input.GetKeyDown(KeyCode.Escape)){
			paused = !paused;
		}
	}
		
		void OnGUI()
		{
		if(paused)
		Pause();
		
		}
	
	void Pause()
	{
		GUILayout.BeginArea(new Rect((Screen.width * 0.5f) -50, (Screen.height * 0.5f) -100, 100, 200));
		if (GUILayout.Button ("Resume"))
			{
			//resume play
			paused = !paused;
			}
		if (GUILayout.Button ("Main Menu"))
			{
			//main menu
			//Application.LoadLevel("Main Menu");
			}
		GUILayout.EndArea();
	}
}

File: Project Playground source code/Assets/Scripts/PauseMenu.js
var paused : boolean = false;
          
function Update () 
{    
if(Input.GetKeyDown(KeyCode.Escape) && paused == false)
	{
	paused = true;
	Time.timeScale = 0;
	}
else if(Input.GetKeyDown(KeyCode.Escape) && paused == true) 
	{
	paused = false;
	Time.timeScale = 1;
	}     
}
